spock-core/src/main/java/org/spockframework/mock/runtime/InteractionScope.java

```diff
@@ -40,26 +41,31 @@ public void addInteraction(final IMockInteraction interaction) {
       final int myRegistrationZone = currentRegistrationZone;
 
       @Override
-      public Object accept(IMockInvocation invocation) {
-        WrongInvocationOrderError wrongInvocationOrderError = null;
+      public Supplier<Object> accept(IMockInvocation invocation) {
+        WrongInvocationOrderError wrongInvocationOrderError;
         if (currentExecutionZone > myRegistrationZone) {
           wrongInvocationOrderError = new WrongInvocationOrderError(decorated, invocation, previousInvocationsInReverseOrder);
+        } else {
+          wrongInvocationOrderError = null;
         }
         currentExecutionZone = myRegistrationZone;
         if (previousInvocationsInReverseOrder.size() == MAX_PREVIOUS_INVOCATIONS) {
           previousInvocationsInReverseOrder.removeLast();
         }
         previousInvocationsInReverseOrder.addFirst(invocation);
 
-        Object result = null;
-        Throwable invocationException = null;
-        try {
-          result = super.accept(invocation);
-        } catch (AssertionError | Exception e) {
-          invocationException = e;
-        }
-        ExceptionUtil.throwWithSuppressed(invocationException, wrongInvocationOrderError);
-        return result;
+        Supplier<Object> supplierFromDecorated = super.accept(invocation);
```

Comment:
Can here any of the calls to `super.accept(invocation)` throw an exception, or assume that the internal supplier code do not throw an exception?
So is here an additional try/catch block missing?

Reply:
There should be no exception thrown.
At least no expected one.
I check through all `accept` methods we have and wrapped all code into the supplied, except for the state-changing code in `InteractionScope` and `MockInteraction` which should be done **within** the `synchronized` scope of the `MockController`.

Actually, in the current state the only think being delegated to here always is a `MockInteraction` anyway if I didn't miss something, so the only things it is doing is adding the invocation to the `acceptedInvocations` and checking the accepted count against the max count.

So I'd not expect any signifiant or important exceptions here and don't think an additional try/catch is necessary.